gr-m2k: Add recipes for libm2k and gr-m2k.

AlexandraTrifan opened the pull request:
This adds a new recipe for the gr-m2k module : https://github.com/analogdevicesinc/gr-m2k and its dependency: libm2k - https://github.com/analogdevicesinc/libm2k

Signed-off-by: AlexandraTrifan <Alexandra.Trifan@analog.com>

willcode reviewed (approved):
Recipes look fine.

FYI: libm2k uses the deprecated `distutils` package, and I ended up with
`Unknown distribution option: 'python_requires'`
When I tried to build.

willcode commented:
It's up to you, I think. PyBOMBS doesn't tend to put requirements on the packages, just the recipes. I just tried to build it in the process of checking out the recipe, and it didn't build with my setup.

My recommendation, though, would be to get rid of any deprecated deps.

AlexandraTrifan commented:
I added a test branch for libm2k which drops distutils and imports setuptools.
I added the test branch in the libm2k recipe. I need to run some more tests for libm2k, in order to check the changes. If everything works fine and this fix works on your setup, I will merge the update to libm2k master branch and point the recipe to that. 
Would that be ok?

willcode commented:
That builds and installs on Fedora 36 now.

PyBOMBS is often run by a non-root user, so you might make sure installation of `53-adi-m2k-usb.rules` is handled or fails gracefully. The PyBOMBS build of libm2k fails due to permissions when run as non-root.

AlexandraTrifan commented:
Regarding my previous comment, after testing the wheel build process that we have in place - I ended up removing just the "python_requires" attribute. A complete migration from distutils is not possible at the moment, since it requires some changes for our entire python wheels procedure. 
I merged this change into the libm2k master branch and updated that in the libm2k.lwr recipe.
Also, the "-DINSTALL_UDEV_RULES=OFF" should skip the UDEV rule (as that can be previously installed by the user).
Do you have some feedback on these changes?

Thank you!
-Alexandra
